templates/stake-pool-template/move/sources/stake_pool.move

```diff
@@ -0,0 +1,519 @@
+        user_stakes: Table<address, UserStake>,
+        // total stake in the contract
+        total_stake: u64,
+        reward_schedule: Option<RewardSchedule>,
+    }
+
+    /// Global per contract
+    struct RewardStoreController has key {
+        extend_ref: ExtendRef,
+    }
+
+    /// Global per contract
+    struct Config has key {
+        // creator can add reward
+        reward_creator: address,
+        // admin can set pending admin, accept admin, update mint fee collector, create FA and update creator
+        admin: address,
+        // pending admin can accept admin
+        pending_admin: Option<address>,
+    }
+
+    /// If you deploy the module under an object, sender is the object's signer
+    /// If you deploy the module under your own account, sender is your account's signer
+    fun init_module(sender: &signer) {
+        let sender_addr = signer::address_of(sender);
+        move_to(sender, Config {
+            reward_creator: @initial_reward_creator_addr,
+            admin: sender_addr,
+            pending_admin: option::none(),
+        });
+
+        let reward_store_constructor_ref = &object::create_object(sender_addr);
+        move_to(sender, RewardStoreController {
+            extend_ref: object::generate_extend_ref(reward_store_constructor_ref),
+        });
+
+        move_to(sender, StakePool {
+            staked_fa_metadata_object: object::address_to_object<Metadata>(@staked_fa_obj_addr),
+            reward_fa_metadata_object: object::address_to_object<Metadata>(@reward_fa_obj_addr),
+            reward_store: fungible_asset::create_store(
+                reward_store_constructor_ref,
+                object::address_to_object<Metadata>(@reward_fa_obj_addr)
+            ),
+            user_stakes: table::new(),
+            total_stake: 0,
+            reward_schedule: option::none(),
+        });
+    }
+
+    // ================================= Entry Functions ================================= //
+
+    /// Set pending admin of the contract, then pending admin can call accept_admin to become admin
+    public entry fun set_pending_admin(sender: &signer, new_admin: address) acquires Config {
+        let sender_addr = signer::address_of(sender);
+        let config = borrow_global_mut<Config>(@stake_pool_addr);
+        assert!(is_admin(config, sender_addr), ERR_ONLY_ADMIN_CAN_SET_PENDING_ADMIN);
+        config.pending_admin = option::some(new_admin);
+    }
+
+    /// Accept admin of the contract
+    public entry fun accept_admin(sender: &signer) acquires Config {
+        let sender_addr = signer::address_of(sender);
+        let config = borrow_global_mut<Config>(@stake_pool_addr);
+        assert!(config.pending_admin == option::some(sender_addr), ERR_ONLY_PENDING_ADMIN_CAN_ACCEPT_ADMIN);
+        config.admin = sender_addr;
+        config.pending_admin = option::none();
+    }
+
+    public entry fun create_reward_schedule(
+        sender: &signer,
+        rps: u64,
```

Comment:
whats `rps`? could we add doc string for this function with explanation of each arguments